packages/blockchain/src/index.ts

```diff
@@ -131,100 +142,149 @@ export default class Blockchain implements BlockchainInterface {
     }
 
     this._heads = {}
-    this._staleHeadBlock = false
-    this._staleHeads = []
 
     this._lock = new Semaphore(1)
-    this._initDone = false
+
+    if (opts.genesisBlock && !opts.genesisBlock.isGenesis()) {
+      throw 'supplied block is not a genesis block'
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.initPromise = this._init(opts.genesisBlock)
```

Comment:
I think I find the pattern we have in the VM now with this [init()](https://github.com/ethereumjs/ethereumjs-vm/blob/master/packages/vm/lib/index.ts#L223) method to be called from outside the constructor more intuitive than this solution, do we eventually want to apply this here as well also for consistency reasons? 🤔

Reply:
I can apply this here, I just think it makes more sense to just let this Promise run in advance and not call this Promise at the point we actually need it to be initialized. But for consistency we can instead use the pattern from VM - let me know what you think 😄

Reply:
I think it's ok as it is now on reconsideration. I think I was just irritated by the external usage of initPromise within the VM, but this was before I realized that this is somewhat of a hack due to the Blockchain initialization in the VM constructor.